addons/l10n_in_sale/controllers/portal.py

```diff
@@ -0,0 +1,25 @@
+from odoo import _
+from odoo.exceptions import ValidationError
+from odoo.addons.sale.controllers import portal as payment_portal
+
+
+class PaymentPortal(payment_portal.PaymentPortal):
+
+    def _create_transaction(self, *args, **kwargs):
```

Comment:
This works better than the previous approaches, but it will only work for transactions created from the portal. Override `create` if you want to cover subscriptions too.

Reply:
Hello @AntoineVDV 
thanks for the response, just wanted to know In sale_subscription [here](https://github.com/odoo/enterprise/blob/master/sale_subscription/controllers/portal.py#L353)
`_create_transaction` function is overrided then why does this will not cover the subscription too.
is I am missing something ?

Reply:
and it inherits the same `PaymentPortal` class
https://github.com/odoo/enterprise/blob/master/sale_subscription/controllers/portal.py#L285

Reply:
`sale_subscription` inherits `_create_transaction` to customize the portal payment flow, but it also adds the cron payment flow that is not covered by `_create_transaction`. However, they both have the `create` method in common.